### PR TITLE
Update doc comments for API functions

### PR DESCRIPTION
**Pull request by jean-roland**

### CODE REVIEW

File: include/zenoh-pico/api/types.h [hunk 1]
```diff
@@ -54,25 +54,24 @@ extern "C" {
...
 /**
- * Represents a variable-length encoding unsigned integer.
+ * Represent a variable-length encoding unsigned integer.
  *
  * It is equivalent to the size of a ``size_t``.
  */
 typedef _z_zint_t z_zint_t;
 
 /**
- * Represents an array of bytes.
```
Comment (sashacmc) [inline include/zenoh-pico/api/types.h:64]: AFAIK in English, when the subject of a sentence is a third-person singular noun or pronoun (he, she, it, or a singular noun like "type"), the verb typically takes an 's' or 'es' at the end in the present tense.
And this comment is essentially an abbreviation for: `The type represents an array of bytes. `

File: include/zenoh-pico/api/types.h [hunk 2]
```diff
@@ -156,9 +149,9 @@ _OWNED_TYPE_PTR(_z_subscriber_t, subscriber)
 _LOANED_TYPE(_z_subscriber_t, subscriber)
 
 /**
- * Represents a Zenoh Publisher entity.
```
Comment (sashacmc) [inline include/zenoh-pico/api/types.h:159]: We talk about class in Zenoh, and class have a name `Publisher`. No?

File: include/zenoh-pico/api/types.h [hunk 3]
```diff
@@ -197,45 +190,42 @@ _OWNED_TYPE_PTR(_z_encoding_t, encoding)
...
  * Members:
- *   z_consolidation_mode_t mode: Defines the consolidation mode to apply to the replies.
+ *   z_consolidation_mode_t mode: the consolidation mode, see :c:type:`z_consolidation_mode_t`
  */
 typedef struct {
     z_consolidation_mode_t mode;
 } z_query_consolidation_t;
 
 /**
- * Represents the set of options that can be applied to a publisher,
```
Comment (sashacmc) [inline include/zenoh-pico/api/types.h:237]: What was the reason to change it?
IMHO main goal of this PR should be a correction for irrelevant comments of those API methods that have been changed.

File: include/zenoh-pico/api/primitives.h [hunk 4]
```diff
@@ -34,134 +34,138 @@ extern "C" {
 /********* Data Types Handlers *********/
 #define z_bytes_wrap _z_bytes_wrap
 
-// TODO(sashacmc): docs
+// TODO(sashacmc): Update doc comments
```
Comment (sashacmc) [inline include/zenoh-pico/api/primitives.h:37]: If you updated the comment why you leave this TODO?

File: include/zenoh-pico/api/primitives.h [hunk 5]
```diff
@@ -34,134 +34,138 @@ extern "C" {
...
  * If given keyexpr contains a declared keyexpr, the resulting owned string will be uninitialized.
  * In that case, the user must use :c:func:`zp_keyexpr_resolve` to resolve the nesting declarations
  * and get its full expanded representation.
  *
  * Parameters:
- *   keyexpr: A loaned instance of :c:type:`z_keyexpr_t`
+ *   keyexpr: Pointer to a loaned instance of :c:type:`z_keyexpr_t`.
+ *   str: Pointer to an uninitialized :c:type:`z_owned_string_t`.
  *
- * Returns:
```
Comment (sashacmc) [inline include/zenoh-pico/api/primitives.h:86]: See my previous comment regarding `s`

File: include/zenoh-pico/api/primitives.h [hunk 6]
```diff
@@ -216,594 +220,472 @@ _Bool z_keyexpr_intersects(const z_loaned_keyexpr_t *l, const z_loaned_keyexpr_t
...
- *   Returns the property with the given integer key from the configuration.
+ * Return:
+ *   The requested property value.
  */
 const char *zp_scouting_config_get(const z_loaned_scouting_config_t *config, uint8_t key);
 
 /**
- * Inserts or replaces the property with the given integer key in the configuration.
```
Comment (sashacmc) [inline include/zenoh-pico/api/primitives.h:384]: `Inserts or replaces` was more relevant, the "update" assumes the key already existed

**Review by sashacmc — CHANGES_REQUESTED** [review]

File: include/zenoh-pico/api/types.h (hunk 1 above)
Comment (jean-roland) [reply include/zenoh-pico/api/types.h:64]: It's kind of the eternal debate between third person and imperative tense. Guess third person is more appropriate here so yeah I'll revert that

**Review by sashacmc — APPROVED** [review]

**Pull request merged**